miniapps/solvers/lor_mms.hpp

```diff
@@ -0,0 +1,95 @@
+#ifndef __LOR_MMS_HPP__
+#define __LOR_MMS_HPP__
+
+extern bool grad_div_problem;
+
+namespace mfem
+{
+
+static constexpr double pi = M_PI, pi2 = M_PI*M_PI;
+
+// Exact solution for definite Helmholtz problem with RHS corresponding to f
+// defined below.
+double u(const Vector &xvec)
+{
+   int dim = xvec.Size();
+   double x = pi*xvec[0], y = pi*xvec[1];
+   if (dim == 2) { return sin(x)*sin(y); }
+   else { double z = pi*xvec[2]; return sin(x)*sin(y)*sin(z); }
+}
+
+double f(const Vector &xvec)
+{
+   int dim = xvec.Size();
+   double x = pi*xvec[0], y = pi*xvec[1];
+
+   if (dim == 2)
+   {
+      return sin(x)*sin(y) + 2*pi2*sin(x)*sin(y);
+   }
+   else // dim == 3
+   {
+      double z = pi*xvec[2];
+      return sin(x)*sin(y)*sin(z) + 3*pi2*sin(x)*sin(y)*sin(z);
+   }
+}
+
+// Exact solution for definite Maxwell and grad-div problems with RHS
+// corresponding to f_vec below.
+void u_vec(const Vector &xvec, Vector &u)
+{
+   int dim = xvec.Size();
+   double x = pi*xvec[0], y = pi*xvec[1];
+   if (dim == 2)
+   {
+      u[0] = cos(x)*sin(y);
+      u[1] = sin(x)*cos(y);
+   }
+   else // dim == 3
+   {
+      double z = pi*xvec[2];
```

Comment:
Since we had to do some work on the projection to support inhomogeneous BC, it seems better to use an exact solution like that in `ex3p` to ensure something doesn't break in the future.

Reply:
Not sure what you mean by this.

Reply:
You can also run with a different geometry, which should allow you to test nonzero tangential components.